experiments/xls/stream_logging.py

```diff
@@ -21,4 +23,43 @@ def __exit__(self, exc_type, exc_value, traceback):
         xls_logger.removeHandler(self.handler)
 
     def get_log(self) -> str:
-        return self._stream.getvalue()
\ No newline at end of file
+        return self._stream.getvalue()
+
+
+class LogParser(object):
+    """
+    A parser that converts raw logs into dicts for ease of use
+    """
+
+    def __init__(self, raw_logs=[]):
+        self.raw_logs = raw_logs
+        self.logs = []
+        self.parse_logs(self.raw_logs)
+
+    def parse_logs(self, raw_logs):
```

Comment:
we don't have a code style guide, but I try to have type annotations everywhere, should we make it a convention? for me, they proved to be quite helpful in both understanding code easier and checking type errors before runtime